Gems/Atom/RHI/DX12/Code/Source/RHI/Scope.cpp

```diff
@@ -101,13 +101,10 @@ namespace AZ
                  D3D12_RESOURCE_STATE_COPY_DEST |
                  D3D12_RESOURCE_STATE_COPY_SOURCE);
 
-            constexpr D3D12_RESOURCE_STATES VALID_GRAPHICS_QUEUE_RESOURCE_STATES =
-                (D3D12_RESOURCE_STATES)DX12_RESOURCE_STATE_VALID_API_MASK;
-
             switch (GetHardwareQueueClass())
             {
             case RHI::HardwareQueueClass::Graphics:
-                return RHI::CheckBitsAll(VALID_GRAPHICS_QUEUE_RESOURCE_STATES, state);
+                return RHI::CheckBitsAll(state, (D3D12_RESOURCE_STATES)DX12_RESOURCE_STATE_VALID_API_MASK);
```

Comment:
Shouldn't this be
return RHI::CheckBitsAll((D3D12_RESOURCE_STATES)DX12_RESOURCE_STATE_VALID_API_MASK), state); and given DX12_RESOURCE_STATE_VALID_API_MASK is 7FFFFFFF maybe we just return true here.